CMakeLists.txt

```diff
@@ -17,21 +17,19 @@ option (IMGUI_IMPL_DX10 "Build the DirectX 10 implementation (only if supported)
 option (IMGUI_IMPL_DX11 "Build the DirectX 11 implementation (only if supported)" ${WIN32})
 option (IMGUI_IMPL_DX12 "Build the DirectX 12 implementation (only if supported)" ${WIN32})
 
-set (CMAKE_CXX_STANDARD 11)
-
-if (IMGUI_DEMO)
-    set (IMGUI_DEMO_SRC imgui_demo.cpp)
-endif ()
-
-add_library (imgui STATIC
+add_library (imgui
     imgui.cpp
     imgui_draw.cpp
     imgui_widgets.cpp
-    ${IMGUI_DEMO_SRC}
+    $<$<BOOL:IMGUI_DEMO>:imgui_demo.cpp>
 )
 
+add_library(imgui::imgui ALIAS imgui)
+
 target_include_directories (imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
 
+set_target_properties(imgui PROPERTIES CXX_STANDARD 11)
```

Comment:
Would using compiler features be better here? I've never quite understood which one was the officially recommended way to do it.

Reply:
Yeah, compiler features are preferred  and idiomatic way.
I realised (thanks to Mike-Devel https://github.com/ocornut/imgui/pull/3027#discussion_r426134699) that setting the CXX_STANDARD property directly is a bad idea as it ignores CMAKE_CXX_STANDARD set by the parent project. Features such as "cxx_std_XX" respect CMAKE_CXX_STANDARD but require cmake 3.8.2 or greater if you want to specify cxx_std_XX instead of specific language features (e.g. cxx_decltype). I initially assumed that you wanted to stick to cmake 3.2 as to not impose undue requirements on consumers.
I can go ahead and amend my pull request in an hour so if you wish to instead use a compiler feature to set cxx_std_11. Furthermore isn't imgui currently only a C++98 library? Should  C++11 be required to consume the imgui target?

Reply:
Yeah that'd be great :)

And I'm not sure of the exact requirement, but at the least we should require c++0x. That's just my opinion though.

Reply:
Well doesn't Imgui currently require only C++98 and that's an currently an advertised aspect of Imgui. Furthermore do you want me to bump the minimum Cmake version to 3.8.2?

Reply:
Yeah you're right:

> Dear ImGui does NOT require C++11 so it can be used with most old C++ compilers.

We should be able to remove it altogether, then. It's up to Imgui to make sure compatibility among archaic compilers, not cmake's, since that is a project choice made by the maintainer.

Good catch :)

What would the reasoning for bumping to 3.8.2 be? I'm not opposed per se, it just implies there's a reason for it :)

Reply:
CMake 3.8.2 is when then `cxx_std_XX` compiler features were introduced. Prior to that, you could only specify what specific language features you required and CMake would use that to figure out what C++ standard version it needed to use on a given compiler.

Reply:
Ah okay, then yes let's bump it and then target `cxx_std_98`.

Reply:
@pumkinpal, https://github.com/pumkinpal/imgui/pull/1